src/node/history.h

```diff
@@ -518,10 +558,7 @@ namespace ccf
     void compact(kv::Version v) override
     {
       flush_pending();
-      if (v > MAX_HISTORY_LEN)
-      {
-        replicated_state_tree.flush(v - MAX_HISTORY_LEN);
-      }
+      replicated_state_tree.flush(v);
```

Comment:
This change is probably unnecessary - in unit tests we can flush the history directly to a specific index, but in the node's real history we were keeping some prior hashes from before the compaction point. I guess this means the signature minitrees include more historical transactions, not just the block to the previous signature?

Reply:
Yes, we should stop doing that, it's not particularly useful.